### ISSUE TITLE
"Cloud" node copies/loops traffic from eth0 to eth1

### ISSUE THREAD

**Issue by Fiveby21** [GNS3/gns3-server#2194]
This has been an ongoing issue for a while. I like to bridge my GNS3 networks into the real world; I don't NAT them, because I want them all to have direct communication with my home network. The problem I'm running into is the way you guys do bridging on the "cloud" node.

I have my GNS3 VM set up with 2 interfaces, both of which connect to VMNet0 (Bridged Network) via VMware Workstation Pro on Windows 11. I have had this same issue in the past with GNS3 on ESXi as well.

* eth0 - Hosts the GNS3 VM's management/control plane; has an IP address of 192.168.1.230
* eth1 - Exists only to bridge the GNS3 environment to the real world. It does not have an IP addressThis is what I connect the "cloud" node to.

Why do I have eth1 at all? Because I don't want all of the GNS3 Client <-> GNS3 VM traffic copied into my topology.

Under normal circumstances, eth1 receives no traffic (other than broadcasts on my home network). I would expect that, once I hook up eth1 (via the Cloud node) to my GNS3 network, that would stay the same. **However**, the moment I connect eth1 into my VM environment, eth0 suddenly starts copying ALL of its network traffic into eth1 (and into my GNS3 topology) and creates a loop. 

Note: in the past, I didn't have eth1; I just connected everything to eth0. Even then, it still created a loop (tons of TCP duplicate errors).

**Before Connecting the Cloud Node:**
![image](https://user-images.githubusercontent.com/40151666/222354347-7d5ab305-4972-4501-83d5-00dc8fbbf1bb.png)

![image](https://user-images.githubusercontent.com/40151666/222354494-2f577fdc-2333-4198-a174-af092cbf9ad5.png)

After Connecting the Cloud Node:

![image](https://user-images.githubusercontent.com/40151666/222355628-2a7838d8-b1d2-4eb4-9107-048fa594f722.png)

Note: The IOU device isn't even turned on. This is JUST the effect of creating the connection between the two nodes.

EDIT: I went ahead and created a bridge (br0) and added eth1 to it to see what would happen.... it created a huge loop, even when not plugged into a cloud node. Seems like there's some weird behavior going on here. My configuration for EVE-NG is identical, and I never have this issue.

**Comment by eantowne** [comment]
This is most likely the problem:

> I have my GNS3 VM set up with 2 interfaces, both of which connect to VMNet0 (Bridged Network) via VMware Workstation Pro on Windows 11.

- Eth0 on the GNS3 VM was designed as a Host-Only interface when used on a desktop Type II hypervisor.
- On ESXi you may have had that issue based on port groups/vswitch promiscuous mode settings.

**Comment by Fiveby21** [comment]
> This is most likely the problem:
> 
> > I have my GNS3 VM set up with 2 interfaces, both of which connect to VMNet0 (Bridged Network) via VMware Workstation Pro on Windows 11.
> 
> * Eth0 on the GNS3 VM was designed as a Host-Only interface when used on a desktop Type II hypervisor.
> * On ESXi you may have had that issue based on port groups/vswitch promiscuous mode settings.

What is the significance of it being deigned as Host-Only? When I used ESXi in the past, I only used Eth0 bridged; I did not have Eth1 in the mix. 

I don't have this issue with EVE-NG.

**Comment by eantowne** [comment]
ESXi and VMware Workstation are 2 different animals.

>What is the significance of it being deigned as Host-Only?
This segregates the "control plane" traffic and does not expose it to the real network or the physical devices physical NIC. It basically creates an isolated virtual switch and connects the VM and a virtual interface on the computer to it.

VMNet0, is also a separate virtual switch, however it bridges to a physical interface on the host system. (actually, I believe it defaults to bridging to ALL of the physical interface on the host computer. So in your case, you have both interfaces (eth0/eth1) plugged into the same switch. Eth1, by virtue of it being a bridged interface is in promiscuous mode and is listening to EVERYTHING.

ESXi, as a Type 1 hypervisor does not have the ability to be on the same host as the desktop client (unless you are doing some black magic deep nested virtualization that you probably should not be doing), thus why it only has an eth0 for the GNS3 VM interfaces. (more can be added, but requires more networking understanding).

These are just my thoughts, and by no means should be taken as absolute truth. Devs are way smarter than I am.

**Comment by Fiveby21** [comment]
Do you mean to say that by adding eth1 to the cloud node, it gets put in promiscuous mode?

**Comment by eantowne** [comment]
In some fashion it would have to be in order for eth1 to listen for traffic destined to other MAC addresses other than its own. I could be wrong, but that seems likely. You could test this by simply changing the eth0 interface configuration in VMware Workstation to "host-only" network and see what the result is.

**Comment by eantowne** [comment]
Additionally, part of the problem is that the way GNS3 has always done the "cloud" node is not actually a Linux bridge, it is using ubridge I believe. which is also why the throughput is normally absolutely terrible. In the future, when the Windows and MacOS "Local Server" goes away, the intent is to switch to a Linux bridge for the "cloud" node, as that is the only OS that node will be running on. (GNS3 VM or a bare metal Linux install).

**Comment by eantowne** [comment]
More information for bridge [HERE](https://github.com/GNS3/ubridge).

**Comment by josephmhiggins** [comment]
@Fiveby21
1. The gns3 developers do not have a "Reset Network Configuration" on the GNS3 VM to return the network configuration as to how they originally designed it.
2. You appear to have changed the default configuration of the GNS3 VM by manually logging into the GNS3 VM and doing something like "I have my GNS3 VM set up with 2 interfaces, both of which connect to VMNet0 (Bridged Network) via VMware Workstation Pro" - that is not a gns3 developer designed network configuration.
3. You have not shown how you changed the configuration. I see no bridge configurations in the GNS3 VM CLI.

This is how it works in engineering....I can very, very, carefully add new behavior to the GNS3 VM all day long.
But I not permitted to change the behavior of the GNS3 VM.
It appears that you have changed the behavior of the GNS3 VM.

My standard design for 7 years was the Bombal design and it looks like this:

![topology](https://user-images.githubusercontent.com/85327995/222619319-a1d3ccd9-ff9e-4f08-932d-8c57c157a9d4.jpg)

After Gloria's post, I upgrade my GNS3 VM network design - without logging into the GNS3 VM AND touching anything in the GNS3 VM to this:

![topology](https://user-images.githubusercontent.com/85327995/222619483-51de5bbc-e7ef-430c-827c-450dbb4ebdcd.jpg)

As you can see in the pngs, I am running wireshark on everything and all packets are where they are supposed to be.

Ean's point is deadly, I think over on the gns3 community board, the default behavior of vmware workstation is to bridge all packets to all interfaces and this caused a major problem in vmware workstation pro 15.1 or 15.2

It is best current practice for a workstation pro user to manually configure the one, and only one bridged interface and not to allow vmware to auto-assign a bridged interface. (VMware workstation 16.2.1 does not allow you to bridge to more than one physical interface)

The VMware workstation pro gui is clunky. Specifically, if i configure the global network editor configuration to bridge to IntelNetworkAdapter2023, then in the GNS3 VM network configuration it shows up as "Auto" - there is nothing I can do about that. That is a deficiency in the VMware gui.

Again, without the REST API to show what I have, my configuration is the Gloria configuration:
vmnet0 bridged
vmnet1 host-only 192.168.25.0./24 (this is for console connections)
vmnet2 host-only 192.168.26.0/24 (this only exists on Workstation Pro and not Workstation Player)
vmnet8 NAT 192.168.27.0/24

Your primary problem is that you: ""I have my GNS3 VM set up with 2 interfaces, both of which connect to VMNet0"

I do not know what to make of that. I do not know what you did.

The default configuration - if you deleted the gns3 vm, uninstalled vmware - in its entirety - except for your license and reinstalled vmware and reinstalled the GNS3 VM will be
vmnet1 host-only aaa.bbb.ccc.ddd/24 (this is for console connections)
vmnet8 NAT eee.fff.ggg.hhh/24

And that configuration will work properly with no packet loops:

The Bombal configuration - if you deleted the gns3 vm, uninstalled vmware - in its entirety - except for your license and reinstalled vmware and reinstalled the GNS3 VM and followed Bombal's instructions: 
https://www.youtube.com/watch?v=Gpje9PV1j8U
vmnet0 bridged
vmnet1 host-only aaa.bbb.ccc.ddd/24 (this is for console connections)
vmnet8 NAT eee.fff.ggg.hhh/24

It will work.

In Windows, if you do not like the VMnet 8 (Nat interface), then control panel - device manger - right click - disable it.

As far as any changes you made to the GNS3 VM by logging in and changing the network configuration - I can provide no assistance.


I recommend
stopping gns3
stopping gns3 vm
start gns3 vm
wave goodbye to all your projects and vms on the gns3 vm - because they will be all wiped out
removing the gns3 vm from vmware
delete the directory that holds the gns3 vm
uninstall vmware
uninstall gns3
install vmware
download a new gns3 vm
install the gns3 vm into vmware
start vmware
start the gns3 vm
install gns3
start gns3

....if you want a bridge interface, follow david's instructions
....if you do not want a nat interface, follow my instructions to disable it via control panel

**Comment by josephmhiggins** [comment]
Oh....I did run the GNS3 VM installed in VMware Workstation Pro, probably version 15.5.1 on my local host about 12-18 months ago. I did not check into the situation you are reporting/worried about.

Are you worried about vmware workstation pro or esxi? I am not getting involved with esxi this month. I will only deal with vmware workstation. I do not believe the gns3 vm, for gns3 version 2.2.38 has bridge-utils (e.g., brctl) installed.

**Comment by josephmhiggins** [comment]
I configured my local gns3 vm was the remote main server.
Everything works as expected with packets only appear where I have designated them to appear on gns3 2.2.37 on windows 10 with vmware workstation pro 16.2.1

 I recommend a new gns3 vm and/or running wireshark somewhere in your remote network before packets reach the gns3 vm.

NTF - no trouble found.

![remove-server-host-nat-bridged](https://user-images.githubusercontent.com/85327995/223200533-36309e3c-0f3a-4a08-b5f0-f95765dee362.JPG)
